waiter/src/waiter/scheduler.clj

```diff
@@ -856,10 +902,16 @@
                          true (conj instance-entry))))))
 
 (defn add-to-store-and-track-failed-instance!
-  [transient-store max-instances-to-keep service-id instance]
-  (log-service-instance instance :fail :info)
-  (add-instance-to-buffered-collection! transient-store max-instances-to-keep service-id instance
-                                        (fn [] #{}) (fn [instances] (-> (sort-instances instances) (rest) (set)))))
+  [transient-store max-instances-to-keep service-id {:keys [id] :as instance}]
+  (let [track-failed-instance-helper
+        (fn track-failed-instance-helper []
+          (do
+            (log-service-instance instance :fail :info)
+            (add-instance-to-buffered-collection! transient-store max-instances-to-keep service-id instance
+                                                  (fn [] #{}) (fn [instances] (-> (sort-instances instances) (rest) (set))))))]
```

Comment:
`(fn [] #{})` → `set`

Reply:
Maybe we should just use a `sorted-set` instead? That would also make the `remove-fn` simpler:
```suggestion
                                                  sorted-set #(disj % (first %)))))]
```

Reply:
This didn't necessarily change in this PR. I only pulled in existing code into a function. Changing to a set may cause downstream effects on other consumers of this value who may be expecting lists.

Reply:
It was already a set — but you're right, this change is unrelated to your PR. I didn't realize this was just whitespace change. I can make a follow-on patch later to make this change separately.